### PR TITLE
fix(iOS): CommandBar Back Button Icon with custom Image

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): fixes https://github.com/unoplatform/nventive-private/issues/129

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Using a custom icon for the CommandBar on iOS the icon is not displayed correctly.

`UIImage` is rendered correctly in place of the BackButton

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by ajpinedam** [comment]
| Before       |  After     |
| :-------------: |  :-----------: |
|   page1_before  |   page1_after   |

File: src/Uno.UI/Controls/CommandBarRenderer.iOS.cs [hunk 1]
```diff
@@ -113,8 +113,9 @@ protected override void Render()
 
 			// CommandBarExtensions.BackButtonIcon
 			var backButtonIcon = Element.GetValue(BackButtonIconProperty) is BitmapIcon bitmapIcon
-				? UIImageHelper.FromUri(bitmapIcon.UriSource)
+				? UIImageHelper.FromUri(bitmapIcon.UriSource)?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal)
```
Comment (davidjohnoliver) [inline src/Uno.UI/Controls/CommandBarRenderer.iOS.cs:116]: Do you understand why this is needed? Is it something that changed between iOS versions?

Comment (ajpinedam) [reply src/Uno.UI/Controls/CommandBarRenderer.iOS.cs:116]: @davidjohnoliver  I believe this is something that changed in iOS11 but I have been looking through the docs and blogs and haven't been able to find it.

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs [hunk 2]
```diff
@@ -232,5 +232,41 @@ public void When_Navigated_CommandBarShouldHideBackButtonTitle_NativeFrame()
...
+			_app.FastTap("NavigateInitialButton");
+
+			_app.WaitForElement("NavigateToPage2Button");
+			_app.FastTap("NavigateToPage2Button");
+
+			_app.WaitForElement("BackButtonImageLoaderButton");
+
+			_app.FastTap("BackButtonImageLoaderButton");
+
+			_app.Wait(TimeSpan.FromMilliseconds(500));
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:257]: What are you waiting for, here? Can you use `WaitForElement()` instead?

**Review by jeromelaban — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs (hunk 2 above)
Comment (ajpinedam) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:257]: @davidjohnoliver  The `FastTap("BackButtonImageLoaderButton");` will get the image from the `BackButton` and place it into the `Image` control that we later use for the validation. This operation takes some time and I want to make sure it has finished before doing the Screenshot

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs [hunk 3]
```diff
@@ -232,5 +232,41 @@ public void When_Navigated_CommandBarShouldHideBackButtonTitle_NativeFrame()
...
+
+			var borderThickness = LogicalToPhysical(3);
+
+			var expectedRect = _app.GetPhysicalRect("RefImage").DeflateBy(borderThickness);
+			var lateRect = _app.GetPhysicalRect("ExpectedImage").DeflateBy(borderThickness);
+
+			ImageAssert.AreAlmostEqual(bmp, expectedRect, bmp, lateRect, permittedPixelError: 10);
+
+			// Removing the global style we added for the CommandBar preventing other UITest to fail
+			_app.FastTap("UnsetGlobalStyleButton");
```
[the remark this answers is not in the document]
Comment (ajpinedam) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/CommandBarTests/UnoSamples_Tests.NativeCommandBar.cs:269]: @jeromelaban  Yes, it is related to this (https://stackoverflow.com/a/34101842/26346).

I have been looking in the Apple documentation about this with no luck but I know this is the case.

**Review by davidjohnoliver — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/3.1

**Pull request merged**